pkg/api/v1alpha1/cluster_types.go

```diff
@@ -290,6 +291,20 @@ type Services struct {
 	CidrBlocks []string `json:"cidrBlocks,omitempty"`
 }
 
+type DNS struct {
+	// ResolvConf refers to the DNS resolver configuration
+	ResolvConf ResolvConf `json:"resolvConf,omitempty"`
+}
+
+type ResolvConf struct {
+	// Path defines the path to the file that contains the DNS resolver configuration
+	Path string `json:"path,omitempty"`
+}
+
+func (n ResolvConf) Equal(o ResolvConf) bool {
```

Comment:
I forget, but since this is not a pointer, can't it just be compared directly through `==`?